homeassistant/components/yardian/sensor.py

```diff
@@ -0,0 +1,153 @@
+"""Sensors for Yardian integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import YardianUpdateCoordinator
+
+
+@dataclass(kw_only=True, frozen=True)
+class YardianSensorEntityDescription(SensorEntityDescription):
+    """Entity description for Yardian sensors."""
+
+    value_fn: Callable[[YardianUpdateCoordinator], StateType]
+
+
+def _rain_delay_value(coordinator: YardianUpdateCoordinator) -> StateType:
+    """Return remaining rain delay in seconds."""
+    oper_info = coordinator.oper_info
+    val = oper_info.get("iRainDelay")
+    if val is None:
+        return None
+
+    try:
+        return max(0, int(val))
+    except (TypeError, ValueError):
+        return None
+
+
+def _active_zone_count_value(coordinator: YardianUpdateCoordinator) -> StateType:
+    """Return number of active zones."""
+    return len(coordinator.device_state.active_zones)
+
+
+def _zone_delay_value(coordinator: YardianUpdateCoordinator) -> StateType:
+    """Return zone delay duration in seconds."""
+    oper_info = coordinator.oper_info
+    val = oper_info.get("iSensorDelay")
+    if val is None:
+        return None
+
+    try:
+        delay = int(val)
+    except (TypeError, ValueError):
+        return None
+
+    if delay > 365 * 24 * 3600:
```

Comment:
Can we assign this number to a constant? I would assume that it's now a year in seconds, but having it as a constant is easier to read